### DISCUSSION TITLE
Async Chunks & Native Components

### DISCUSSION THREAD

**Discussion by steventnorris** [callstack/repack#448] [Q&A] [answered]
I know that Re:Pack supports Module Federation and Async Chunks, but what is the line as far as needing to redeploy the application to store vs being able to use a CDN to, at runtime, download the Async Chunk? Specifically, wondering how this works with the native component compilation in RN and remotely serving those chunks. 

Note: Not related to store reviews. I'm aware of the iOS store review issues with remote deployed code updates.

**Comment by steventnorris** [comment]
According to MS CodePush (which does a very similar thing), a recompile and new store/MDM push is only needed if it touches native code or adds new native dependencies. In their README, they state:

> Note: Any product changes which touch native code (e.g. modifying your AppDelegate.m
> */*MainActivity.java
> file, adding a new plugin) cannot be distributed via CodePush, and therefore, must be updated via the appropriate store(s).

Which reads to me like as long as you don't add native dependencies or modify native code, you're good to push from CDN and not have to recompile and push through store. If anyone else has experience with this and can confirm or deny, please do.

**Comment by jbroma** [comment, marked as answer]
> According to MS CodePush (which does a very similar thing), a recompile and new store/MDM push is only needed if it touches native code or adds new native dependencies. In their README, they state:
> 
> > Note: Any product changes which touch native code (e.g. modifying your AppDelegate.m
> > */*MainActivity.java
> > file, adding a new plugin) cannot be distributed via CodePush, and therefore, must be updated via the appropriate store(s).
> 
> Which reads to me like as long as you don't add native dependencies or modify native code, you're good to push from CDN and not have to recompile and push through store. If anyone else has experience with this and can confirm or deny, please do.

Hi @steventnorris,

this is pretty much spot on, you can keep redeploying with CDN as long as your changes are limited to JS, including adding new JS-only dependencies.

**Reply by steventnorris** [reply]
@jbroma The part I'm still having a little difficulty drawing a line in the sand on is UI components. By my understanding, RN compiles UI components into native code and uses a jsbridge to link functionality to the js code. If I were to, for example, change the layout of a component by adding new View, Text, etc and removing others, would that also be considered a native code change or is that still deployable via remote chunks through a CDN? (not touching any native components directly, only altering the RN code)

**Reply by jbroma** [reply]
That's actually not how it works - the JS executes on the device and it creates new native views for these components in runtime (broadly speaking). You can change the JS in runtime, so to answer your question directly - changing layout inside of React components would not be considered a native change :)

**Reply by steventnorris** [reply]
Ah that makes a lot more sense, and is something I should have known. Didn't dig deeply into the "native functionality" portion of the js bridge as I should have there. Thank you for the help! Oddly, this isn't well documented centrally anywhere that I could find in relation to loading async chunks in RN, hence why I was pilfering things from the CodePush docs. That's a better situation than I thought it was!

**Reply by jbroma** [reply]
Glad I could help! [Here's](https://reactnative.dev/docs/intro-react-native-components#native-components) more on the subject if you are interested how RN works under the hood!
